zeth-contracts/contracts/BaseMixer.sol

```diff
@@ -126,47 +256,42 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
 
         // 2. We re-assemble the nullifiers (JSInputs) and check they were not already seen
         bytes32[jsIn] memory nfs;
-        uint256[] memory digest_inputs = new uint[](2);
-        for(uint i = 1; i < 1 + 2*jsIn; i += 2) {
-            digest_inputs[0] = primary_inputs[i];
-            digest_inputs[1] = primary_inputs[i+1];
-            nfs[(i-1)/2] = Bytes.sha256_digest_from_field_elements(digest_inputs);
+        for(uint i; i < jsIn; i++) {
+            nfs[i] = assemble_nullifier(i, primary_inputs);
             require(
-                !nullifiers[nfs[(i-1)/2]],
+                !nullifiers[nfs[i]],
                 "Invalid nullifier: This nullifier has already been used"
             );
-            nullifiers[nfs[(i-1)/2]] = true;
+            nullifiers[nfs[i]] = true;
         }
 
         // 3. We re-compute h_sig, re-assemble the expected h_sig and check they are equal
         // (i.e. that h_sig re-assembled was correctly generated from vk)
         bytes32 expected_hsig = sha256(abi.encodePacked(nfs, vk));
-
-        digest_inputs[0] = primary_inputs[1 + 2 * (jsIn + jsOut) + 1 + 1];
-        digest_inputs[1] = primary_inputs[1 + 2 * (jsIn + jsOut + 1) + 1];
-        bytes32 hsig = Bytes.sha256_digest_from_field_elements(digest_inputs);
+        bytes32 hsig = assemble_hsig(primary_inputs);
         require(
             expected_hsig == hsig,
             "Invalid hsig: This hsig does not correspond to the hash of vk and the nfs"
         );
     }
 
+
     function assemble_commitments_and_append_to_state(uint[] memory primary_inputs) internal {
         // We re-assemble the commitments (JSOutputs)
-        uint256[] memory digest_inputs = new uint[](2);
-        for(uint i = 1 + 2 * jsIn ; i < 1 + 2*(jsIn + jsOut); i += 2) {
-            digest_inputs[0] = primary_inputs[i]; // See the way the inputs are ordered in the extended proof
-            digest_inputs[1] = primary_inputs[i+1];
-            bytes32 current_commitment = Bytes.sha256_digest_from_field_elements(digest_inputs);
-            uint commitment_address = insert(current_commitment);
-            emit LogCommitment(commitment_address, current_commitment);
+        for(uint i ; i < jsOut; i ++) {
```

Comment:
`for(uint i ; i < jsOut; i ++)` -> `for (uint i; i < jsOut; i++)`

Reply:
Fixed.